helios-retrofit-integration/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt

```diff
@@ -0,0 +1,17 @@
+package helios.retrofit
+
+import arrow.core.flatMap
+import helios.core.Json
+import helios.typeclasses.Decoder
+import okhttp3.ResponseBody
+import retrofit2.Converter
+import java.nio.ByteBuffer
+import java.nio.CharBuffer
+
+class HeliosResponseBodyConverter<T>(private val decoder: Decoder<T>) : Converter<ResponseBody, T> {
+  override fun convert(value: ResponseBody): T? = Json
+    .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
+    .flatMap { it.decode(decoder) }
+    .fold({ null }, { it })
```

Comment:
`getOrElse { null }`?

Reply:
Yes you are right. I usually forget there are more functions than fold 😅

Reply:
All the implementations lead to fold